lib/spack/spack/cmd/concretize.py

```diff
@@ -4,8 +4,16 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import spack.environment as ev
+import llnl.util.tty.color as clr
 
-description = 'concretize an environment and write a lockfile'
+description = clr.colorize(
```

Comment:
The description is supposed to be a short one-line summary. I would add the extra info you want to `subparser.epilog` so it gets displayed in the help message. Note that `spack spec -I` uses the same glyphs, so you may want to add documentation there too.

Reply:
Install and buildcache commands use it too, although build cache uses a different `status_fn` so the meanings are totally different. 
I moved the glyph definitions to class attributes so they can be used directly in the help output, and there's no continuity issues if they ever change again.
I also documented what sort of function is expected for the `status_fn`, and changed Spec.install_status to be consistent with that. There's only one instance of a different `status_fn` being used, and it only kind-of happens to work. 

It's a bit broken that the 'upstream' status is calculated independent of that function, but for the moment that doesn't matter.